src/js/pass/deterministic.ts

```diff
@@ -10,6 +10,7 @@ import {Trigger} from '../rom/trigger.js';
 import {hex} from '../rom/util.js';
 import {assert} from '../util.js';
 import { Monster } from '../rom/monster.js';
+import * as Patterns from '../rom/pattern.js';
```

Comment:
(nit) we only use UpperCamelCase for constructors, not namespaces, so this should be `patterns` (or `pattern`).  Or would it make sense to move these to a `newtiles` file?